src/modules/CRUD/components/Details/Edit/Edit.js

```diff
@@ -70,18 +94,46 @@ class Edit extends React.Component {
     } = this.props;
     // Remove controlDraw from controls
     updateControls([]);
-    if (this.layerId) {
+    if (this.layerId && Object.keys(map).length) {
       map.setFilter(this.layerId, ['all']);
     }
   }
 
   get layerId () {
-    const { layerPaint: { id } = {} } = this.props;
+    const { layerPaint: { id } } = this.props;
     return id;
   }
 
+  initEdit = () => {
+    const {
+      schema,
+      feature: { geom = {} },
+      action,
+    } = this.props;
+
+    this.setState({
+      schema,
+      geom,
+    });
+    if (action !== ACTION_UPDATE || Object.keys(geom).length) {
+      this.initDraw();
```

Comment:
Shouldn't this be as [second argument of `setState`](https://fr.reactjs.org/docs/react-component.html#setstate)?